packages/flutter/lib/src/material/text_field.dart

```diff
@@ -188,6 +188,65 @@ class _TextFieldSelectionGestureDetectorBuilder extends TextSelectionGestureDete
 /// ```
 /// {@end-tool}
 ///
+/// ## Reading values
+///
+/// A common way to read a value from a TextField is to use the [onSubmitted]
+/// callback. This callback is applied to the text field's current value when
+/// the user finishes editing.
+///
+/// {@tool dartpad --template=stateful_widget_material}
+///
+/// This sample shows how to get a value from a TextField via the [onSubmitted]
+/// method.
+///
+/// ```dart
+/// TextEditingController _controller;
+///
+/// void initState() {
+///   _controller = TextEditingController();
+/// }
+///
+/// void dispose() {
+///   _controller.dispose();
+/// }
+///
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Center(
+///       child: TextField(
+///         controller: _controller,
+///         onSubmitted: (String value) async {
+///           await showDialog<void>(
+///             context: context,
+///             builder: (BuildContext context) {
+///               return AlertDialog(
+///                 title: const Text('Thanks!'),
+///                 content: Text ('You typed "$value".'),
+///                 actions: <Widget>[
+///                   FlatButton(
+///                     onPressed: () { Navigator.pop(context); },
+///                     child: const Text('OK'),
+///                   ),
+///                 ],
+///               );
+///             },
+///           );
+///         },
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// In addition to [onSubmitted], [onEditingComplete] is also called when the
+/// user finishes editing. It differs in that passing [onEditingComplete]
+/// prevents the default behavior of focus and submission of changes to the
```

Comment:
This doesn't really make the use case for onEditingComplete clear. I think it would be worth explaining that, by default, when the user finishes editing the text controller's text value is updated, the keyboard focus is given up (unless the user hits "next" or "previous" key like tab and shift-tab). The onEditingComplete callback can overrdie the default behavior. 

For most apps and most cases, the onSubmitted callback is sufficient